torch/_inductor/ir.py

```diff
@@ -7005,11 +7047,20 @@ def create(cls, x: IRNode, device: torch.device, non_blocking: bool) -> IRNode:
 
         developer_warning("DeviceCopy in input program")
         constant_args = (non_blocking,)
+        is_destination_pinned = (
+            x_device.type == "cuda" and device.type == "cpu" and non_blocking
+        )
+        is_source_pinned = (
+            x_device.type == "cpu" and device.type == "cuda" and non_blocking
```

Comment:
this is not necessarily true, e.g. `cuda_tensor.to("cpu", non-blocking=True)` will always allocate destination in pinned memory, but `cpu_tensor.to("cuda", non_blocking=True)` might or might not be pinned depending on what user was doing

Reply:
agreed, if cpu_tensor is allocated outside of inductor, we can't control this. but it seems setting the tensor to pinned if it is controller by inductor is a good idea?